agent/agent.go

```diff
@@ -21,13 +21,41 @@ import (
 	"github.com/google/gops/signal"
 )
 
-// Start starts the gops agent on a host process. Once agent started,
-// users can use the advanced gops features.
+// Agent represents an agent that enable the advanced gops features
+type Agent struct {
```

Comment:
This type was necessary both to return it on a `Listen()` call (allowing `Close()` to be called) and to allow changing config flags when using functional options.
Returning a private type in the `Listen()` call has the awkward side-effect of allowing things like:
```
a, err := agent.Listen()
```
But disallowing:
```
var a agent.Agent
a, err = agent.Listen()
````